### DISCUSSION TITLE
Weighted EMA

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#1039] [Ideas]
Hi Dave,

hope you ar doing well.

can we add Weighted EMA to our library ..?

Thanks

**Comment by DaveSkender** [comment]
Already have it for you: [WMA](https://dotnet.stockindicators.dev/indicators/Wma/#content)

**Reply by DaveSkender** [reply]
Oh wait, you said EMA, that’s potentially different.

**Reply by DaveSkender** [reply]
If it’s this [EWMA](https://corporatefinanceinstitute.com/resources/capital-markets/exponentially-weighted-moving-average-ewma/), looks like a good one to add to the backlog.  Looks like a regular EMA, but with a custom weighting factor.

**Reply by DaveSkender** [reply]
for EMA `alpha = 2/(n+1)` where here `alpha` is user specified.  If you think about, it’s the exact same formula.  If you pre-calculate `n = 2/alpha - 1`, it’d work with current EMA method.

**Reply by DaveSkender** [reply]
this might be a simple alt API to the same underlying calculation

**Reply by myalgomate** [reply]
Yes got your point, looks fine.
